react/features/toolbox/components/AbstractToolboxItem.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import { Component } from 'react';
+
+type Styles = {
+
+    /**
+     * Style for the item's icon.
+     */
+    iconStyle: Object,
+
+    /**
+     * Style for the item itself.
+     */
+    style: Object,
+
+    /**
+     * Color for the item underlay (shows when clicked).
+     */
+    underlayColor: string
+};
+
+export type Props = {
+
+    /**
+     * A succinct description of what the item does. Used by accessibility
+     * tools and torture tests.
+     */
+    accessibilityLabel: string,
+
+    /**
+     * Whether this item is disabled or not. When disabled, clicking an the item
+     * has no effect, and it may reflect on its style.
+     */
+    disabled: boolean,
+
+    /**
+     * The name of the icon of this {@code ToolboxItem}.
+     */
+    iconName: string,
+
+    /**
+     * The text associated with this item. When `showLabel` is set to
+     * {@code true}, it will be displayed alongside the icon.
+     */
+    label: string,
```

Comment:
At this point in time all labels have the same value as the tooltip. If we envision those 2 ever being different we can keep it that way, otherwise we could fold it into a single field, maybe.